src/EtlLib.Nodes.AmazonS3/AmazonS3WriterEtlOperation.cs

```diff
@@ -67,30 +74,43 @@ public override IEnumerableEtlOperationResult<AmazonS3WriterResult> ExecuteWithR
                 _awsCredentials = new BasicAWSCredentials(context.Config[Constants.S3AccessKeyId], context.Config[Constants.S3SecretAccessKey]);
 
             var results = new List<AmazonS3WriterResult>();
-            using (var client = new AmazonS3Client(_awsCredentials, _awsRegionEndpoint))
+            using (var client = new TransferUtility(new AmazonS3Client(_awsCredentials, _awsRegionEndpoint)))
             {
                 foreach (var file in _files)
                 {
                     var objectKey = Path.GetFileName(file);
 
-                    var request = new PutObjectRequest()
+                    var request = new TransferUtilityUploadRequest()
                     {
-
                         BucketName = _bucketName,
                         Key = objectKey,
                         FilePath = file,
                         StorageClass = _storageClass
                     };
+                    startTime = DateTime.Now;
+                    progress = 0;
+                    request.UploadProgressEvent +=
+                        uploadRequest_UploadPartProgressEvent;
 
-                    var result = client.PutObjectAsync(request).GetAwaiter().GetResult();
-
-                    results.Add(new AmazonS3WriterResult(objectKey, result));
+                    client.UploadAsync(request).Wait();
+                    //results.Add(new AmazonS3WriterResult(objectKey, result));     

```

Comment:
You actually change behavior here because there is no longer a resultset containing the resulting object id's which could be used for further processing.  I have some changes that will mitigate this and still achieve what you're looking for.